pkg/controller/atlasdeployment/atlasdeployment_controller_test.go

```diff
@@ -767,182 +770,274 @@ func TestUniqueKey(t *testing.T) {
 }
 
 func TestReconciliation(t *testing.T) {
-	t.Run("should reconcile with existing cluster", func(t *testing.T) {
-		secret := &corev1.Secret{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      "api-secret",
-				Namespace: "default",
-				Labels: map[string]string{
-					"atlas.mongodb.com/type": "credentials",
-				},
-			},
-			Data: map[string][]byte{
-				"orgId":         []byte("1234567890"),
-				"publicApiKey":  []byte("a1b2c3"),
-				"privateApiKey": []byte("abcdef123456"),
+	secret := &corev1.Secret{
```

Comment:
for better isolation of each test case you need, I'd not update this test but add new cases starting with an existing cluster and test only search node parts.
A single unit test should no be used to test all paths inside the reconciliation flow, it will become unmaintainable.